nexus/src/app/sagas/instance_create.rs

```diff
@@ -299,11 +306,245 @@ impl NexusSaga for SagaInstanceCreate {
             )?;
         }
 
+        builder.append(configure_asic_action());
         builder.append(instance_ensure_action());
         Ok(builder.build()?)
     }
 }
 
+async fn sic_add_network_config(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    let params = sagactx.saga_params::<Params>()?;
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
+    let osagactx = sagactx.user_data();
+    let dpd_client: Arc<dpd_client::Client> =
+        Arc::clone(&osagactx.nexus().dpd_client);
+    let datastore = &osagactx.datastore();
+    let log = sagactx.user_data().log();
+
+    let sled_uuid = sagactx.lookup::<Uuid>("server_id")?;
+
+    let instance_name = sagactx.lookup::<db::model::Name>("instance_name")?;
+    let (.., authz_instance, db_instance) = LookupPath::new(&opctx, &datastore)
+        .project_id(params.project_id)
+        .instance_name(&instance_name)
+        .fetch()
+        .await
+        .map_err(ActionError::action_failed)?;
+
+    let instance_id = db_instance.id();
+
+    let (.., sled) = LookupPath::new(&opctx, &datastore)
+        .sled_id(sled_uuid)
+        .fetch()
+        .await
+        .map_err(ActionError::action_failed)?;
+
+    let sled_ip_address = sled.address();
+
+    debug!(log, "fetching network interfaces");
+
+    // TODO: Correctness
+    // Currently we're only mapping external ip addresses to the first
+    // network interface we find.We'll need a more sophisticated
+    // approach as we expand our use cases.
+    let network_interface = datastore
+        .derive_guest_network_interface_info(&opctx, &authz_instance)
+        .await
+        .map_err(ActionError::action_failed)?
+        .first()
+        .ok_or_else(|| {
+            ActionError::action_failed(Error::internal_error(
+                "guest network interface not found",
+            ))
+        })?
+        .clone();
+
+    // Is there a cleaner way to get from sled_agent_types::MacAddr -> MacAddr6
+    // (and ultimately dpd_client::types::MacAddr)?
```

Comment:
Not right now. I want to make an `oxide-net` crate that has a bunch of these types we've duplicated in a billions places. But until then, nope.